clusterloader2/pkg/test/simple_test_executor.go

```diff
@@ -117,14 +117,19 @@ func (ste *simpleExecutor) ExecuteTestSteps(ctx Context, conf *api.Config) *erro
 		return errors.NewErrorList(fmt.Errorf("automanaged namespaces creation failed: %v", err))
 	}
 
+	steps, err := flattenModuleSteps(ctx, conf.Steps)
```

Comment:
Nit: For the random code reader (like me), it's not clear why conf.Steps up to this point had type '[]api.Step', but after this line we start working on '[]*api.Step'.

I think it's error-prone that for loop below expects pointers, while half of this function uses pointers and half of this function uses objects directly.

Can we keep []api.Step everywhere (i.e. change return type for flattenModuleSteps)?

I guess, some day we can remove the flattenModuleSteps, without modifying loop below and we will have a bug in code.

Reply:
Would it change anything in terms of readability? We're operating on pointers in the methods below the call stack (e.g. executeStep). I'm just making the conversion step before. It's reasonable to operate on pointers from the beginning as we're avoiding unnecessary copying of steps (which we'd have if flattenModuleSteps returned []api.Step).

I don't buy the argument about the bug. If you made that change the code wouldn't compile.

Reply:
Makes sense. I haven't noticed that ExecuteStep expects pointer. In this case it will stop compiling. Otherwise, the code below would compile perfectly well, but will work incorrectly.